src/radical/saga/adaptors/noop/noop_job.py

```diff
@@ -0,0 +1,683 @@
+        self.monitor = _job_state_monitor(self._logger)
+        self.monitor.start()
+
+        return self.get_api()
+
+
+    # --------------------------------------------------------------------------
+    #
+    def close(self):
+
+        if self.monitor:
+            self.monitor.stop()
+
+
+    # --------------------------------------------------------------------------
+    #
+    #
+    def _job_run(self, job):
+        '''
+        runs a job on the wrapper via pty, and returns the job id
+        '''
+
+        if not job.jd.executable.endswith('sleep'):
+            raise ValueError('expected "sleep", not %s' % job.jd.executable)
+
+
+        if len(job.jd.arguments) != 1:
+            raise ValueError('expected int argument, not %s' % job.jd.arguments)
+
+        job.tgt = time.time() + int(job.jd.arguments[0])
+        self.monitor.add_job(job)
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def run_job(self, cmd, host):
+        '''
+        Implements adaptors.cpi.job.Service.run_job()
+        '''
+
+        if not cmd:
+            raise rse.BadParameter._log(self._logger,
+                    "run_job needs a command to run")
+
+        if host and host != self.rm.host:
+            raise rse.BadParameter._log(self._logger,
+                    "Can only run jobs on %s, not on %s" % (self.rm.host, host))
+
+        jd = api.Description()
+
+        exe, arg = cmd.split()
+
+        jd.executable = exe
+        jd.arguments  = [arg]
+
+        job = self.create_job(jd)
+        job.run()
+
+        return job
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def create_job(self, jd):
+
+        # this dict is passed on to the job adaptor class -- use it to pass any
+        # state information you need there.
+        adaptor_state = {"job_service"    : self,
+                         "job_description": jd,
+                         "job_schema"     : self.rm.schema }
+
+        return api.Job(_adaptor=self._adaptor, _adaptor_state=adaptor_state)
+
+
+    # --------------------------------------------------------------------------
+    @SYNC_CALL
+    def get_url(self):
+
+        return self.rm
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def list(self):
+
+        return self.monitor.list_jobs()
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def get_job(self, job_id, no_reconnect=False):
+
+        return self.monitor.get_job(job_id)
+
+
+    # --------------------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def container_run(self, jobs):
+        '''
+        From all the job descriptions in the container, build a bulk, and submit
+        as async.  The read whaterver the wrapper returns, and sort through the
```

Comment:
The read -> Then read

Reply:
Thanks, that comment was off anyway - fixed.